scipy/spatial/transform/rotation.py

```diff
@@ -657,3 +657,105 @@ def __mul__(self, other):
         if self._single and other._single:
             result = result[0]
         return self.__class__(result, normalized=True)
+
+    def __getitem__(self, val):
+        # Returns a single rotation for an index, and a stack of rotation(s)
+        # for slice, index array, etc even if only one index is ultimately
+        # given
+        return self.__class__.from_quaternion(self._quat[val], normalized=True)
+
+    def copy(self):
+        return self.__class__.from_quaternion(
+            self.as_quaternion(), normalized=True)
+
+
+class Slerp(object):
+    """Spherical Linear Interpolation of Rotations.
+
+    The interpolation between consecutive rotations is performed as a rotation
+    around a fixed axis with a constant angular velocity [1]_. This ensures
+    that the interpolated rotations follow the shortest path between initial
+    and final orientations.
+
+    Parameters
+    ----------
+    t : array_like, shape (N,)
+        Times of the known rotations. At least 2 times must be specified.
+    r : `Rotation` instance
+        `N` rotations to perform the interpolation between. At least 2
+        rotations must be specified.
+
+    Methods
+    -------
+    __call__
+
+    References
+    ----------
+    .. [1] `Quaternion Slerp
+            <https://en.wikipedia.org/wiki/Slerp#Quaternion_Slerp>`_
+    """
+    def __init__(self, t, r):
+        # We don't need to store r, just r[:-1] and interpolation rotvecs
+        if r.num_rots == 1:
+            raise ValueError("Expected at least 2 rotations in object, "
+                             "got {}.".format(r.num_rots))
+
+        t = np.asarray(t)
+        if t.ndim != 1:
+            raise ValueError("Expected times to be specified in a 1 "
+                             "dimensional array, got {} "
+                             "dimensions.".format(t.ndim))
+
+        if t.shape[0] != r.num_rots:
+            raise ValueError("Expected number of rotations to be equal to "
+                             "number of timestamps given, got {} rotations "
+                             "and {} timestamps.".format(
+                                r.num_rots, t.shape[0]))
+        self.times = t
+        self.timedelta = np.diff(t)
+
+        if np.any(self.timedelta <= 0):
+            raise ValueError("Times must be in strictly increasing order.")
+
+        # inv() and __getitem__ return new objects anyway
+        self.rots = r[:-1]
+        # should be equal to r.rots - 1
+        self.num_rots = self.rots.num_rots
+        rot_params = self.rots.inv() * r[1:]
```

Comment:
This name is hard to understand. Perhaps the best idea is not to introduce any intermediate variables.